async-nats/src/options.rs

```diff
@@ -180,17 +180,16 @@ impl ConnectOptions {
     /// ```no_run
     /// # #[tokio::main]
     /// # async fn main() -> Result<(), async_nats::ConnectError> {
-    /// let nc = async_nats::ConnectOptions::with_token("t0k3n!".into())
+    /// let nc = async_nats::ConnectOptions::new()
+    ///     .with_token("t0k3n!".into())
     ///     .connect("demo.nats.io")
     ///     .await?;
     /// # Ok(())
     /// # }
     /// ```
-    pub fn with_token(token: String) -> Self {
-        ConnectOptions {
-            auth: Authorization::Token(token),
-            ..Default::default()
-        }
+    pub fn with_token(mut self, token: String) -> Self {
```

Comment:
After digesting it for a bit, I don't like this breaking change.

1. It's a breaking change
2. It makes API inconsistent - all other options are without `with`.

So we break api and do not get good api.

My suggestion would be to leave those method as they are, and add relevant `token`, `user_and_password` etc.

That way we have consistency across options and we don't break anything.

In next step we might deprecate the `with` methods, but we don't HAVE to.

@caspervonb cc